bundles/org.openhab.binding.hdpowerview/src/test/java/org/openhab/binding/hdpowerview/HDPowerViewJUnitTests.java

```diff
@@ -502,42 +502,201 @@ public void testDualRailConstraints() {
-        assertEquals(new PercentType(75), shade.getState(caps, PRIMARY_ZERO_IS_CLOSED));
+        shade.setPosition(caps, SECONDARY_POSITION, 0).setPosition(caps, PRIMARY_POSITION, 100);
+        shade.setPosition(caps, SECONDARY_POSITION, 50).setPosition(caps, PRIMARY_POSITION, 75);
+        assertEquals(new PercentType(50), shade.getState(caps, SECONDARY_POSITION));
+        assertEquals(new PercentType(75), shade.getState(caps, PRIMARY_POSITION));
 
         // ==== IMPOSSIBLE !! primary at middle, secondary below => test the constraining code ====
-        shade.setPosition(caps, SECONDARY_ZERO_IS_OPEN, 0).setPosition(caps, PRIMARY_ZERO_IS_CLOSED, 100);
-        shade.setPosition(caps, PRIMARY_ZERO_IS_CLOSED, 60).setPosition(caps, SECONDARY_ZERO_IS_OPEN, 75);
-        assertEquals(new PercentType(60), shade.getState(caps, PRIMARY_ZERO_IS_CLOSED));
-        assertEquals(new PercentType(60), shade.getState(caps, SECONDARY_ZERO_IS_OPEN));
+        shade.setPosition(caps, SECONDARY_POSITION, 0).setPosition(caps, PRIMARY_POSITION, 100);
+        shade.setPosition(caps, PRIMARY_POSITION, 60).setPosition(caps, SECONDARY_POSITION, 75);
+        assertEquals(new PercentType(60), shade.getState(caps, PRIMARY_POSITION));
+        assertEquals(new PercentType(60), shade.getState(caps, SECONDARY_POSITION));
 
         // ==== OK !! primary at middle, secondary above ====
-        shade.setPosition(caps, SECONDARY_ZERO_IS_OPEN, 0).setPosition(caps, PRIMARY_ZERO_IS_CLOSED, 100);
-        shade.setPosition(caps, PRIMARY_ZERO_IS_CLOSED, 60).setPosition(caps, SECONDARY_ZERO_IS_OPEN, 25);
-        assertEquals(new PercentType(60), shade.getState(caps, PRIMARY_ZERO_IS_CLOSED));
-        assertEquals(new PercentType(25), shade.getState(caps, SECONDARY_ZERO_IS_OPEN));
+        shade.setPosition(caps, SECONDARY_POSITION, 0).setPosition(caps, PRIMARY_POSITION, 100);
+        shade.setPosition(caps, PRIMARY_POSITION, 60).setPosition(caps, SECONDARY_POSITION, 25);
+        assertEquals(new PercentType(60), shade.getState(caps, PRIMARY_POSITION));
+        assertEquals(new PercentType(25), shade.getState(caps, SECONDARY_POSITION));
+    }
+
+    /**
+     * Test parsing of DuoLite shades having a secondary blackout shade.
+     *
+     */
+    @Test
+    public void testDuoliteShadePositionParsing() {
+        ShadePosition test;
+        State pos;
+
+        // blackout shades have capabilities 8
+        Capabilities capabilities = db.getCapabilities(8);
+
+        // both shades up
+        test = new ShadePosition().setPosition(capabilities, PRIMARY_POSITION, 0);
+        assertNotNull(test);
+        pos = test.getState(capabilities, PRIMARY_POSITION);
+        assertEquals(PercentType.class, pos.getClass());
+        assertEquals(0, ((PercentType) pos).intValue());
+        pos = test.getState(capabilities, SECONDARY_POSITION);
+        assertTrue(PercentType.ZERO.equals(pos));
```

Comment:
Consider introducing a private method `assertShadePosition()` or similar for reducing redundancy.

Reply:
Ok